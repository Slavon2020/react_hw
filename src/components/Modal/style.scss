.modal {
	width: 400px;
	width: 520px;
	margin: 0 auto;
	position: fixed;
	left: calc(50% - 200px);
	top: 40%;
	z-index: 10;
	color: white;
	background: #e74c3c;
	box-sizing: border-box;
	border-radius: 10px;
		&-header {
			background: #484b5d;
			display: flex;
			justify-content: space-between;
			padding: 5px 10px;
			border-top-left-radius: 5px;
			border-top-right-radius: 5px;
		}
		&-body {
			background: white;
			padding:  0 40px 10px 40px;
			border-bottom-left-radius: 5px;
			border-bottom-right-radius: 5px;
			color: black;
		}
		&-text {
			padding: 10px;
		}
		&-buttons {
			display: flex;
			justify-content: center;
			& span {
				margin-right: 10px;
			}
		}
}
.overlay {
	background: rgba(0, 0, 0, 0.3);
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	z-index: 5;
}