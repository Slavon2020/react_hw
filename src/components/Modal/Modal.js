import React, {Fragment} from 'react';
import Button from '../Button/Button';
import './style.scss'
import propTypes from 'prop-types'


function Modal(props) {
	const {header, text, closeButton, actions, toggleAddToCartModal} = props;
	return (
		<Fragment>
			<div onClick={toggleAddToCartModal} className='overlay'></div>
			<div className='modal'>
				<div className='modal-header'>
					<span className='modal-header-text'>{header}</span>
					{closeButton && <Button className={'close-btn'} onClick={toggleAddToCartModal} text='X'/>}
				</div>
				<div className='modal-body'>
					<p className='modal-text'>{text}</p>
					<div className='modal-buttons'>
						{actions.map((el, index) => {
							return (
								<span key={index}>
									{el}
									</span>
							)
						})}
					</div>
				</div>
			</div>
		</Fragment>
	);
}

Modal.propTypes = {
	header: propTypes.string.isRequired,
	text: propTypes.string.isRequired,
	closeButton: propTypes.bool,
	actions: propTypes.array.isRequired,
	toggleAddToCartModal: propTypes.func.isRequired
}

Modal.defaultProps = {
	closeButton: false
}

export default Modal;