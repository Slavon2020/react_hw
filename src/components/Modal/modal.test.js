import Modal from './Modal';
import { render, screen } from '@testing-library/react';

const modalText = 'test-modal-text';
const modalHeaderText = 'test-modal-header-text';

const mockFunc = jest.fn();

describe('testing Modal.js', () => {
	test('smoke test Modal.js', () => {
		render(<Modal toggleAddToCartModal={mockFunc} actions={[]} text={modalText} header={modalHeaderText}/>);
		screen.getByText(modalText);
		screen.getByText(modalHeaderText);
	});

	test('testing by snapshot in Modal.js', () => {
		const { container } =  render(<Modal toggleAddToCartModal={mockFunc} actions={[]} text={modalText} header={modalHeaderText}/>);
		expect(container).toMatchSnapshot()
	})
})