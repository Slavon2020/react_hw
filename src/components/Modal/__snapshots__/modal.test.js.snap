// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Modal.js testing by snapshot in Modal.js 1`] = `
<div>
  <div
    class="overlay"
  />
  <div
    class="modal"
  >
    <div
      class="modal-header"
    >
      <span
        class="modal-header-text"
      >
        test-modal-header-text
      </span>
    </div>
    <div
      class="modal-body"
    >
      <p
        class="modal-text"
      >
        test-modal-text
      </p>
      <div
        class="modal-buttons"
      />
    </div>
  </div>
</div>
`;
